internal/manifests/targetallocator/configmap.go

```diff
@@ -94,6 +94,9 @@ func ConfigMap(params Params) (*corev1.ConfigMap, error) {
 			prometheusCRConfig["scrape_interval"] = taSpec.PrometheusCR.ScrapeInterval.Duration
 		}
 
+		prometheusCRConfig["allow_namespaces"] = taSpec.PrometheusCR.AllowNamespaces
+		prometheusCRConfig["deny_namespaces"] = taSpec.PrometheusCR.DenyNamespaces
```

Comment:
It'd make things easier if we only set these when they were non-empty. Then we wouldn't need to update all the tests.